x/evm/keeper/statedb_test.go

```diff
@@ -476,9 +475,43 @@ func (suite *KeeperTestSuite) TestEmpty() {
 }
 
 func (suite *KeeperTestSuite) TestSnapshot() {
+
+	var key = common.BytesToHash([]byte("key"))
+	var value1 = common.BytesToHash([]byte("value1"))
+	var value2 = common.BytesToHash([]byte("value2"))
+
+	suite.SetupTest()
+
+	// set to the initial context
+	suite.app.EvmKeeper.SetState(suite.address, key, value1)
+	suite.Require().Equal(value1, suite.app.EvmKeeper.GetState(suite.address, key))
+
 	revision := suite.app.EvmKeeper.Snapshot()
 	suite.Require().Zero(revision)
-	suite.app.EvmKeeper.RevertToSnapshot(revision) // no-op
+
+	suite.app.EvmKeeper.SetState(suite.address, key, value2)
+	suite.Require().Equal(value2, suite.app.EvmKeeper.GetState(suite.address, key))
+
+	suite.app.EvmKeeper.RevertToSnapshot(revision)
+
+	// reverted
+	suite.Require().Equal(value1, suite.app.EvmKeeper.GetState(suite.address, key))
+
+	// test a nested snapshot/revert
+	revision1 := suite.app.EvmKeeper.Snapshot()
+	suite.app.EvmKeeper.SetState(suite.address, key, value2)
+
+	revision2 := suite.app.EvmKeeper.Snapshot()
+
+	// delete
+	suite.app.EvmKeeper.SetState(suite.address, key, common.Hash{})
+	suite.Require().Equal(common.Hash{}, suite.app.EvmKeeper.GetState(suite.address, key))
+
+	suite.app.EvmKeeper.RevertToSnapshot(revision2)
+	suite.Require().Equal(value2, suite.app.EvmKeeper.GetState(suite.address, key))
+
+	suite.app.EvmKeeper.RevertToSnapshot(revision1)
+	suite.Require().Equal(value1, suite.app.EvmKeeper.GetState(suite.address, key))
```

Comment:
I'd also add a test case that reverts directly to revision1 instead of doing `revision2 -> revision1`

Reply:
done